# Descriptive Statistics of the US Poverty Rate from 2018 to 2022
An exploratory data analysis of the National Database of Childcare Prices 2022 dataset to understand some key socio-economic variables accross the States in the United States of America and their influence on poverty rate levels in those States.

## 1.0 Overview

**Key Objectives of the Analysis**
* Analyse the US unemployment rate of males and females across 5 years (2018 to 2022).
* Analyse the unemployment rate per state and their relationship to the poverty rate levels in 2022.
* Analyse the median household income per State against the poverty rate per State in 2022.
* Analyse the effects of houseolds with only the father working compared to the both parents working on the poverty rate in the States in 2022.
* Analyse the median weekly family childcare and centre-based cares price against the poverty rate per US State in 2022.

__Key Variables__  
1) Unemploymet rate
2) Median household income per State
3) Number of houseolds with only the father working
4) Number of houseolds with both parents working
5) Median weekly centre-based care price
6) Median weekly Family childcare price
7) Poverty rate

## 2.0 Data Import and Preparation

### 2.1 Importing the Data and Python Packages Necessary
**Data Source**  
The National Database of Childcare Prices (NDCP) under the United States Department of Labour. https://dataportal.dol.gov/datasets/10278

In [296]:
#import necessary libraries and packages
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import warnings

In [297]:
warnings.filterwarnings("ignore") #to avoid clogging the notebook with warnings, they will be filtered

In [298]:
#import the raw data
raw_df = pd.read_csv("./dataset/NDCP_2022_dataset.csv")
raw_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 48308 entries, 0 to 48307
Columns: 370 entries, state_name to ifemp_p_state
dtypes: float64(200), int64(167), str(3)
memory usage: 136.4 MB


In [299]:
raw_df.describe().T

,count,mean,std,min,25%,50%,75%,max
county_fips_code,48308.0,31394.875362,16291.031838,1001.0,19032.5,30025.0,46107.0,72153.0
studyyear,48308.0,2014.999959,4.320660,2008.0,2011.0,2015.0,2019.0,2022.0
emr_16,48308.0,54.860845,8.743072,11.0,49.5,55.6,61.2,98.3
femr_16,48308.0,50.968001,7.994420,13.6,46.0,51.4,56.5,100.0
memr_16,48308.0,58.979668,10.430401,8.0,52.9,60.2,66.4,100.0
...,...,...,...,...,...,...,...,...
imemp_n_state,48308.0,1.000000,0.000000,1.0,1.0,1.0,1.0,1.0
ifemp_n_state,48308.0,1.000000,0.000000,1.0,1.0,1.0,1.0,1.0
iemp_p_state,48308.0,1.000000,0.000000,1.0,1.0,1.0,1.0,1.0
imemp_p_state,48308.0,1.000000,0.000000,1.0,1.0,1.0,1.0,1.0


In [300]:
raw_df.head()

,state_name,state_abbreviation,county_name,county_fips_code,studyyear,emr_16,femr_16,memr_16,emr_20to64,femr_20to64,...,ifemp_service_state,iemp_sales_state,imemp_sales_state,ifemp_sales_state,iemp_n_state,imemp_n_state,ifemp_n_state,iemp_p_state,imemp_p_state,ifemp_p_state
0,Alabama,AL,Autauga County,1001,2008,61.5,56.1,67.3,71.4,66.3,...,1,1,1,1,1,1,1,1,1,1
1,Alabama,AL,Autauga County,1001,2009,60.6,54.8,67.0,72.5,66.9,...,1,1,1,1,1,1,1,1,1,1
2,Alabama,AL,Autauga County,1001,2010,60.4,54.9,66.4,72.3,67.4,...,1,1,1,1,1,1,1,1,1,1
3,Alabama,AL,Autauga County,1001,2011,58.7,52.4,65.6,71.0,64.9,...,1,1,1,1,1,1,1,1,1,1
4,Alabama,AL,Autauga County,1001,2012,57.7,52.2,63.8,70.7,65.1,...,1,1,1,1,1,1,1,1,1,1


### 2.2 Processing and Preparing the Dataset for Analysis

The dataset contains 48,308 observations and 370 attributes. Half of the attributes are metadata attributes that specify whether data was systematically imputed for a particular missing attribute of an observation e.g. `iemr_16_state` is a binary data that specifies whether the 'employment rate of females in the county' for an observation was systematically imputed. These metadata columns and other attributes with low level of relevance to the key objectives of analysis will be removed.

Since the analysis focuses on the period of 2018 to 2022, only the observations captured at that period will be retained.

The relevant attributes are:  

| Field Name | Field Description | Intended Data Type |
|------------|------------------|--------------------|
| state_name | Name of state for which the data are being entered. | categorical
| state_abbreviation | Two-letter state abbreviation | categorical
| county_name | Name of the county for which data are being entered. | categorical
| county_fips_code | Five-digit number that uniquely identifies the county in a state. | categorical
| studyyear | Year in which data were collected for the market rate survey. | categorical
| unr_16 | Unemployment rate of the population aged 16 years old or older | numeric |
| funr_16 | Unemployment rate of the female population aged 16 years old or older | numeric |
| munr_16 | Unemployment rate of the male population aged 16 years old or older | numeric |
| mhi_2022 | Median household income expressed in 2022 dollars | numeric |
| h_under6_fwork | Number of households with children under 6 years old with only the father working | numeric |
| h_under6_mwork | Number of households with children under 6 years old with only the mother working | numeric |
| h_under6_singlem | Number of households with children under 6 years old with a single mother | numeric |
| h_6to17_fwork | Number of households with children between 6 and 17 years old with only the father working | numeric |
| h_6to17_mwork | Number of households with children between 6 and 17 years old with only the mother working | numeric |
| h_6to17_singlem | Number of households with children between 6 and 17 years old with a single mother | numeric |
| h_under6_bothwork | Number of households with children under 6 years old with two working parents | numeric |
| h_6to17_bothwork | Number of households with children between 6 and 17 years old with two working parents | numeric |
| mcbto5 | Weekly, full-time median price charged for Center-Based Care for those aged 0 to 5 months | numeric |
| mc6to11 | Weekly, full-time median price charged for Center-Based Care for those aged 6 to 11 months | numeric |
| mc12to17 | Weekly, full-time median price charged for Center-Based Care for those aged 12 to 17 months | numeric |
| mc18to23 | Weekly, full-time median price charged for Center-Based Care for those aged 18 to 23 months | numeric |
| mc24to29 | Weekly, full-time median price charged for Center-Based Care for those aged 24 to 29 months | numeric |
| mc30to35 | Weekly, full-time median price charged for Center-Based Care for those aged 30 to 35 months | numeric |
| mc36to41 | Weekly, full-time median price charged for Center-Based Care for those aged 36 to 41 months | numeric |
| mc42to47 | Weekly, full-time median price charged for Center-Based Care for those aged 42 to 47 months | numeric |
| mc48to53 | Weekly, full-time median price charged for Center-Based Care for those aged 48 to 53 months | numeric |
| mc54tosa | Weekly, full-time median price charged for Center-Based Care for those aged 54 months to school age | numeric |
| mcsa | Weekly, full-time median price charged for Center-Based Care for those who are school age | numeric |
| mfccbto5 | Weekly, full-time median price charged for Family Childcare for those aged 0 to 5 months | numeric |
| mfcc6to11 | Weekly, full-time median price charged for Family Childcare for those aged 6 to 11 months | numeric |
| mfcc12to17 | Weekly, full-time median price charged for Family Childcare for those aged 12 to 17 months | numeric |
| mfcc18to23 | Weekly, full-time median price charged for Family Childcare for those aged 18 to 23 months | numeric |
| mfcc24to29 | Weekly, full-time median price charged for Family Childcare for those aged 24 to 29 months | numeric |
| mfcc30to35 | Weekly, full-time median price charged for Family Childcare for those aged 30 to 35 months | numeric |
| mfcc36to41 | Weekly, full-time median price charged for Family Childcare for those aged 36 to 41 months | numeric |
| mfcc42to47 | Weekly, full-time median price charged for Family Childcare for those aged 42 to 47 months | numeric |
| mfcc48to53 | Weekly, full-time median price charged for Family Childcare for those aged 48 to 53 months | numeric |
| mfcc54tosa | Weekly, full-time median price charged for Family Childcare for those aged 54 months to school age | numeric |
| mfccsa | Weekly, full-time median price charged for Family Childcare for those who are school age | numeric |
| pr_p | Poverty rate for individuals | numeric |

**Filtering for the Relevant Columns and rows**

In [301]:
#filter for the relevant variables
filtered_df = raw_df[["state_name", "state_abbreviation","county_name", "county_fips_code", "studyyear", "unr_16", "funr_16", "munr_16", "mhi_2022", "h_under6_fwork", "h_under6_mwork", "h_under6_singlem", "h_6to17_fwork", "h_6to17_mwork", "h_6to17_singlem", "h_under6_bothwork", "h_6to17_bothwork", "mcbto5", "mc6to11", "mc12to17", "mc18to23", "mc24to29", "mc30to35", "mc36to41", "mc42to47", "mc48to53", "mc54tosa", "mcsa", "mfccbto5", "mfcc6to11", "mfcc12to17", "mfcc18to23", "mfcc24to29", "mfcc30to35", "mfcc36to41", "mfcc42to47", "mfcc48to53", "mfcc54tosa", "mfccsa", "pr_p"]]
filtered_df = filtered_df[filtered_df["studyyear"] >= 2018] #filter only data from 2018 to 2022
filtered_df = filtered_df.reset_index(drop=True) #reset the indexes so that the dataset is arranged well
filtered_df.head()

,state_name,state_abbreviation,county_name,county_fips_code,studyyear,unr_16,funr_16,munr_16,mhi_2022,h_under6_fwork,...,mfcc12to17,mfcc18to23,mfcc24to29,mfcc30to35,mfcc36to41,mfcc42to47,mfcc48to53,mfcc54tosa,mfccsa,pr_p
0,Alabama,AL,Autauga County,1001,2018,4.2,3.4,4.9,69367,1332,...,112.33,112.33,111.58,110.93,110.93,110.93,110.93,110.93,107.79,15.4
1,Alabama,AL,Autauga County,1001,2019,3.7,3.2,4.1,68128,1003,...,117.55,117.55,116.05,115.62,115.62,115.62,115.62,115.62,110.20,15.2
2,Alabama,AL,Autauga County,1001,2020,2.9,2.4,3.3,66679,837,...,122.78,122.78,120.53,120.31,120.31,120.31,120.31,120.31,112.60,15.2
3,Alabama,AL,Autauga County,1001,2021,2.8,2.3,3.3,68299,855,...,128.00,128.00,125.00,125.00,125.00,125.00,125.00,125.00,115.00,13.6
4,Alabama,AL,Autauga County,1001,2022,2.8,2.6,2.9,68315,833,...,133.23,133.23,129.48,129.69,129.69,129.69,129.69,129.69,117.40,11.4


In [302]:
filtered_df.shape #show the shape of the dataset

(16103, 40)

By focusing only on variables and rows that are relevant to this analysis project, the total number of relvant rows is 16,103 and the number of variables is 40.

**Treating blanks**  
Checking the rows for blanks

In [303]:
print("List of columns and their number of blanks rows:")
filtered_df.isna().sum()

List of columns and their number of blanks rows:


state_name               0
state_abbreviation       0
county_name              0
county_fips_code         0
studyyear                0
unr_16                   0
funr_16                  0
munr_16                  0
mhi_2022                 0
h_under6_fwork           0
h_under6_mwork           0
h_under6_singlem         0
h_6to17_fwork            0
h_6to17_mwork            0
h_6to17_singlem          0
h_under6_bothwork        0
h_6to17_bothwork         0
mcbto5                2639
mc6to11               2639
mc12to17              2627
mc18to23              2627
mc24to29              2618
mc30to35              2618
mc36to41              2602
mc42to47              2602
mc48to53              2602
mc54tosa              2602
mcsa                  2673
mfccbto5              3019
mfcc6to11             3019
mfcc12to17            3018
mfcc18to23            3018
mfcc24to29            3026
mfcc30to35            3026
mfcc36to41            3014
mfcc42to47            3014
mfcc48to53            3014
m

The blank rows of each attribute for each State will be replaced by the mean of that attribute for that State alone instead of the general attribute's column to avoid filling the blanks of some State with values that may be larger than average due to some states with high variable values.

In [304]:
states_abvs = filtered_df["state_abbreviation"].value_counts().index #getting the list of states' abbreviations

In [ ]:
#creating a reusable function for filling all blank rows in all the columns of a dataframe with their respective means
def fill_blanks_with_mean(series):
    if(series.dtype != "string"): #this is to skip string-type columns to avoid having error when the mean is being calculated
        n_series = series.fillna(series.mean())
        return n_series
    else:
        return series

In [ ]:
#creating a reusable function for filling all blank rows in all the columns of a dataframe with their respective medians
def fill_blanks_with_median(series):
    if(series.dtype != "string"): #this is to skip string-type columns to avoid having error when the median is being calculated
        n_series = series.fillna(series.median())
        return n_series
    else:
        return series

In [ ]:
#replacing blanks in each subset of the data for each State with their respective averages
for state_ID in states_abvs:
    filtered_df[filtered_df["state_abbreviation"] == state_ID] = filtered_df[filtered_df["state_abbreviation"] == state_ID].apply(fill_blanks_with_mean)

In [308]:
def show_cols_with_blanks(df):
    count = 0
    for col in filtered_df.columns:
        blank_count = df[col].isna().sum()
        if(blank_count > 0):
            count += 1
            print(f"'{col}' contains {blank_count} blanks")

    if(count == 0):
        print("There are no more blank rows")

In [309]:
show_cols_with_blanks(filtered_df) #confirming if any column still contain blanks, if there is any, show them

'mcbto5' contains 625 blanks
'mc6to11' contains 625 blanks
'mc12to17' contains 625 blanks
'mc18to23' contains 625 blanks
'mc24to29' contains 625 blanks
'mc30to35' contains 625 blanks
'mc36to41' contains 625 blanks
'mc42to47' contains 625 blanks
'mc48to53' contains 625 blanks
'mc54tosa' contains 625 blanks
'mcsa' contains 630 blanks
'mfccbto5' contains 625 blanks
'mfcc6to11' contains 625 blanks
'mfcc12to17' contains 625 blanks
'mfcc18to23' contains 625 blanks
'mfcc24to29' contains 625 blanks
'mfcc30to35' contains 625 blanks
'mfcc36to41' contains 625 blanks
'mfcc42to47' contains 625 blanks
'mfcc48to53' contains 625 blanks
'mfcc54tosa' contains 625 blanks
'mfccsa' contains 630 blanks


Checking the names of the States that still have blank rows

In [310]:
print("The names of states still having blanks:")
filtered_df[filtered_df["mc18to23"].isna()]["state_name"].value_counts() #check the names of the States with blanks

The names of states still having blanks:


state_name
Indiana       460
New Mexico    165
Name: count, dtype: int64

The States of Indiana and New Mexico appears to not have any values inputed for all categories of __"child care prices"__, these will be filled with the national **median** from all the States. The median is used instead of the mean to avoid influencing the values with high variables/outliers from some States.

In [311]:
filtered_df = filtered_df.apply(fill_blanks_with_median) #fill all remaining blank rows in each attribute with the national median of each attribute

In [312]:
show_cols_with_blanks(filtered_df) #check if there is still any blank rows

There are no more blank rows


In [313]:
#checking the observations from the State of Indiana again
filtered_df[filtered_df["state_abbreviation"] == "IN"].head()

,state_name,state_abbreviation,county_name,county_fips_code,studyyear,unr_16,funr_16,munr_16,mhi_2022,h_under6_fwork,...,mfcc12to17,mfcc18to23,mfcc24to29,mfcc30to35,mfcc36to41,mfcc42to47,mfcc48to53,mfcc54tosa,mfccsa,pr_p
3488,Indiana,IN,Adams County,18001,2018,3.6,3.7,3.6,61021,1850,...,130.943333,129.25,124.765,124.446165,122.505,122.5,121.115,121.075,109.2,16.8
3489,Indiana,IN,Adams County,18001,2019,3.6,3.7,3.5,60905,1775,...,130.943333,129.25,124.765,124.446165,122.505,122.5,121.115,121.075,109.2,16.8
3490,Indiana,IN,Adams County,18001,2020,3.0,3.1,3.0,60619,1744,...,130.943333,129.25,124.765,124.446165,122.505,122.5,121.115,121.075,109.2,16.5
3491,Indiana,IN,Adams County,18001,2021,3.0,3.3,2.8,61080,1749,...,130.943333,129.25,124.765,124.446165,122.505,122.5,121.115,121.075,109.2,15.3
3492,Indiana,IN,Adams County,18001,2022,3.2,3.0,3.4,61731,1946,...,130.943333,129.25,124.765,124.446165,122.505,122.5,121.115,121.075,109.2,11.5


In [314]:
filtered_df.describe().T

,count,mean,std,min,25%,50%,75%,max
county_fips_code,16103.0,31388.104639,16293.520829,1001.00,19031.000,30023.000000,46105.000000,72153.00
studyyear,16103.0,2020.000186,1.414236,2018.00,2019.000,2020.000000,2021.000000,2022.00
unr_16,16103.0,5.543433,3.277965,0.00,3.700,5.000000,6.600000,38.40
funr_16,16103.0,5.278271,3.449434,0.00,3.300,4.700000,6.400000,39.00
munr_16,16103.0,5.777284,3.692781,0.00,3.700,5.200000,7.000000,46.50
mhi_2022,16103.0,61618.497050,17608.384925,14013.00,50924.000,59926.000000,69707.500000,170935.00
h_under6_fwork,16103.0,1616.255294,5738.645421,0.00,129.000,340.000000,999.000000,169014.00
h_under6_mwork,16103.0,148.142085,510.559053,0.00,3.000,29.000000,104.000000,14769.00
h_under6_singlem,16103.0,1837.049183,6458.804549,0.00,159.000,447.000000,1168.000000,205199.00
h_6to17_fwork,16103.0,2736.802770,10264.146612,0.00,225.000,579.000000,1651.000000,327471.00
